x-pack/elastic-agent/pkg/core/composable/providers.go

```diff
@@ -0,0 +1,30 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package composable
+
+import "context"
+
+// FetchContextProvider is the interface that a context provider uses so as to be able to be called
+// explicitely on demand by vars framework in order to fetch specific target values like a k8s secret.
+type FetchContextProvider interface {
+	// Run runs the inventory provider.
+	Run(ContextProviderComm) error
```

Comment:
Being this is a subset of the `ContextProvider`, I think it would be better to define it as:

```
type FetchContextProvider interface {
    ContextProvider

    // Run runs the inventory provider.
    Fetch(string) (string, error)
}
```